### PR TITLE
[OSDEV-1126] - Implement OS ID search (BE)

### PR DESCRIPTION
**Pull request by mazursasha1990**
[OSDEV-1126](https://opensupplyhub.atlassian.net/browse/OSDEV-1126) - SLC. Implement OS ID search (BE).

Added the `historical_os_id` field to the response from the `v1/production-locations` endpoint if the searched production location contains this data. Modified the search query for `os_id` so that the search is conducted in both the `os_id` and `historical_os_id` fields in the OpenSearch production-locations index.
To make this possible, the `sync_production_locations.sql` script, which generates data for the production-locations index, was modified to include the selection of `historical_os_id_value` from the `api_facilityalias` table.
Additionally, a `historical_os_id` filter was added to the `sync_production_locations.conf`, ensuring that the `historical_os_id` is included in the index document only when the `historical_os_id_value` is not empty.

[OSDEV-1126]: https://opensupplyhub.atlassian.net/browse/OSDEV-1126?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/logstash/indexes/production_locations.json [hunk 1]
```diff
@@ -184,6 +184,14 @@
               "type": "keyword"
             }
           }
+        },
+        "historical_os_id": {
```
Comment (VadimKovalenkoSNF) [inline src/logstash/indexes/production_locations.json:188]: Use plural name such as `historical_os_id` since this is the array of items.

**Review by VadimKovalenkoSNF — CHANGES_REQUESTED** [review]
Left question / few comments.

File: src/django/api/tests/test_opensearch_query_builder.py [hunk 2]
```diff
@@ -40,13 +40,62 @@ def test_add_multi_match(self):
...
         self.builder.add_terms('country', ['US', 'CA'])
         expected = {'terms': {'country.alpha_2': ['US', 'CA']}}
         self.assertIn(
             expected,
             self.builder.query_body['query']['bool']['must']
-            )
+        )
+
+    def test_add_terms_for_os_id(self):
+        self.builder.add_terms('os_id', ['1', '2'])
```
[the remark this answers is not in the document]
Comment (mazursasha1990) [reply src/django/api/tests/test_opensearch_query_builder.py:52]: Changed.

**Review by VadimKovalenkoSNF — APPROVED** [review]
Approved. Pls, ask @nazarkolossandf to go over all v1 endpoints after meging this PR.

**Review by roman-stolar — APPROVED** [review]
LGTM

**Pull request merged**